lib/python/qmk/cli/doctor.py

```diff
@@ -153,14 +150,18 @@ def is_executable(command):
         return False
 
     # Make sure the command can be executed
-    check = subprocess.run([command, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5, universal_newlines=True)
+    check = subprocess.run([command, '-dumpversion'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5, universal_newlines=True)
+
+    if check.returncode > 1:  # if -dumpversion returns error check with --version instead
+        check = subprocess.run([command, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5, universal_newlines=True)
+
```

Comment:
```suggestion
    version_arg = ESSENTIAL_BINARIES[command].get('version_arg', '--version')
    check = run([command, version_arg], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5, universal_newlines=True)
```
I can't add a suggestion to the `ESSENTIAL_BINARIES` dict, as it's outside of the changed hunks, but it's just:
```
    'avr-gcc': {'version_arg': '-dumpversion'},
    'arm-none-eabi-gcc': {'version_arg': '-dumpversion'},
```